### ISSUE TITLE
Limit 0 or null as parameter should return all records

### ISSUE THREAD

**Issue by dszymczuk** [rethinkdb/rethinkdb#5807]
Example (I use rethinkdbdash):

`r.table(tableNotification).orderBy(r.desc('time')).limit(limit)`
is better than:
```
if(limit > 0){
	r.table(tableNotification).orderBy(r.desc('time')).limit(limit).run()
} else {
	r.table(tableNotification).orderBy(r.desc('time')).run()
}
```

After comment @danielmewes in https://github.com/rethinkdb/docs/issues/1133 I think, that null will be good to get all records.
Also nth and skip should the same option.

**Comment by danielmewes** [comment]
Thanks for re-opening the issue here.

I'm marking this as a ReQL_proposal. We'll discuss it for one of the upcoming releases.

**Comment by VeXocide** [comment]
I agree with @danielmewes that we shouldn't special case `0` and have `limit(0)` return zero documents, that's what you've asked for after all. Adding support for `limit(null)` on the other hand sounds reasonable.

Not sure what `nth(null)` would do though, and I'm not sure there's much use for a `skip(null)` that skips all documents either.

**Comment by tatsujin1** [comment]
I think, at least `limit(null)` and `skip(null)` have intuitive English meanings; "no limit" and "skip nothing"... 
However, `nth(null)` is less intuitive... if it even makes any sense?

**Comment by dszymczuk** [comment]
I would like to focus on limit. Because when I want to use limit, I have to use construction with "if" as above.
It's not comfortable :)

You're right - nth doesn't make sense.
